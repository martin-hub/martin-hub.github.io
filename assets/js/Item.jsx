import React from 'react';
import styled from 'styled-components';
import {media} from './media.jsx';

const LogoImage = styled.img``;

const TileImage = styled.img`
  display: block;
  width: 100%;
`;

const ItemName = styled.div`
  color: #FEFEFC;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
`;

const ItemBottomBar = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  height: 28%;
  background-color: rgba(0, 0, 0, 0.5);

  ${LogoImage} {
    flex: 0 1;
    height: 80%;
    padding: 0 16px;
  }

  ${ItemName} {
    flex: auto;

    ${media.xxlarge`
      font-size: 55px;
    `}
    
    ${
  media.xlarge`
      font-size: 45px;
    `}
    
    ${media.large`
      font-size: 38px;
    `}

    ${media.medium`
      font-size: 30px;
    `}

    ${media.small`
      font-size: 20px;
    `}
  }
`;

const ItemTile = styled.div`
  position: relative;
`;

const ItemContainer = styled.div`
  position: relative;
  background-color: black;
  width: 900px;
  max-width: 100%;
  padding: 16px;

  ${media.medium`
    width: 75%;
  `}

  ${media.small`
    width: 100%;
    padding: 10px;
  `}

  ${ItemTile} {
    height: 100%;
  }

  ${ItemBottomBar} {
    position: absolute;
    right: 0;
    bottom: 0;
    left: 0;
  }
`;

/**
 * This component puts together the individual styled sub-component, to present an item box.
 */
const Item = ({className, title, imageUrl, ...rest}) => <ItemContainer>
  <a href='#' title={title} className={className} {...rest}>
    <ItemTile>
      <TileImage src={imageUrl}/>
      <ItemBottomBar>
        <LogoImage src='image/logo.png'/>
        <ItemName>{title}</ItemName>
      </ItemBottomBar>
    </ItemTile>
  </a>
</ItemContainer>;

export default Item;
