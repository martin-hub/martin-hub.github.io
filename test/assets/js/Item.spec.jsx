import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Item from 'Item.jsx';

describe("Item", () => {
  test('renders correctly', () => {
    expect(renderer.create(<Item />).toJSON()).toMatchSnapshot();
    expect(renderer.create(<Item title="Rick and Morty" image="image/rnm.jpg" />).toJSON()).toMatchSnapshot();
    expect(renderer.create(<Item className="test-class" otherFakeAttr='fake-value' />).toJSON()).toMatchSnapshot();
  });
});