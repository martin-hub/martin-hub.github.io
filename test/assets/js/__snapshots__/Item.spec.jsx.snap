// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item renders correctly 1`] = `
.c3 {
  display: block;
  width: 100%;
}

.c8 {
  color: #FEFEFC;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 28%;
  background-color: rgba(0,0,0,0.5);
}

.c5 .c6 {
  -webkit-flex: 0 1;
  -ms-flex: 0 1;
  flex: 0 1;
  height: 80%;
  padding: 0 16px;
}

.c5 .c7 {
  -webkit-flex: auto;
  -ms-flex: auto;
  flex: auto;
}

.c2 {
  position: relative;
}

.c0 {
  position: relative;
  background-color: black;
  width: 900px;
  max-width: 100%;
  padding: 16px;
}

.c0 .c1 {
  height: 100%;
}

.c0 .c4 {
  position: absolute;
  right: 0;
  bottom: 0;
  left: 0;
}

@media (min-width:120.0625em) {
  .c5 .c7 {
    font-size: 55px;
  }
}

@media (max-width:120em) {
  .c5 .c7 {
    font-size: 45px;
  }
}

@media (max-width:90em) {
  .c5 .c7 {
    font-size: 38px;
  }
}

@media (max-width:75.625em) {
  .c5 .c7 {
    font-size: 30px;
  }
}

@media (max-width:47.9375em) {
  .c5 .c7 {
    font-size: 20px;
  }
}

@media (max-width:75.625em) {
  .c0 {
    width: 75%;
  }
}

@media (max-width:47.9375em) {
  .c0 {
    width: 100%;
    padding: 10px;
  }
}

<div
  className="c0"
>
  <a
    className={undefined}
    href="#"
    title={undefined}
  >
    <div
      className="c1 c2"
    >
      <img
        className="c3"
        src={undefined}
      />
      <div
        className="c4 c5"
      >
        <img
          className="c6 "
          src="image/logo.png"
        />
        <div
          className="c7 c8"
        />
      </div>
    </div>
  </a>
</div>
`;

exports[`Item renders correctly 2`] = `
.c3 {
  display: block;
  width: 100%;
}

.c8 {
  color: #FEFEFC;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 28%;
  background-color: rgba(0,0,0,0.5);
}

.c5 .c6 {
  -webkit-flex: 0 1;
  -ms-flex: 0 1;
  flex: 0 1;
  height: 80%;
  padding: 0 16px;
}

.c5 .c7 {
  -webkit-flex: auto;
  -ms-flex: auto;
  flex: auto;
}

.c2 {
  position: relative;
}

.c0 {
  position: relative;
  background-color: black;
  width: 900px;
  max-width: 100%;
  padding: 16px;
}

.c0 .c1 {
  height: 100%;
}

.c0 .c4 {
  position: absolute;
  right: 0;
  bottom: 0;
  left: 0;
}

@media (min-width:120.0625em) {
  .c5 .c7 {
    font-size: 55px;
  }
}

@media (max-width:120em) {
  .c5 .c7 {
    font-size: 45px;
  }
}

@media (max-width:90em) {
  .c5 .c7 {
    font-size: 38px;
  }
}

@media (max-width:75.625em) {
  .c5 .c7 {
    font-size: 30px;
  }
}

@media (max-width:47.9375em) {
  .c5 .c7 {
    font-size: 20px;
  }
}

@media (max-width:75.625em) {
  .c0 {
    width: 75%;
  }
}

@media (max-width:47.9375em) {
  .c0 {
    width: 100%;
    padding: 10px;
  }
}

<div
  className="c0"
>
  <a
    className={undefined}
    href="#"
    image="image/rnm.jpg"
    title="Rick and Morty"
  >
    <div
      className="c1 c2"
    >
      <img
        className="c3"
        src={undefined}
      />
      <div
        className="c4 c5"
      >
        <img
          className="c6 "
          src="image/logo.png"
        />
        <div
          className="c7 c8"
        >
          Rick and Morty
        </div>
      </div>
    </div>
  </a>
</div>
`;

exports[`Item renders correctly 3`] = `
.c3 {
  display: block;
  width: 100%;
}

.c8 {
  color: #FEFEFC;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 28%;
  background-color: rgba(0,0,0,0.5);
}

.c5 .c6 {
  -webkit-flex: 0 1;
  -ms-flex: 0 1;
  flex: 0 1;
  height: 80%;
  padding: 0 16px;
}

.c5 .c7 {
  -webkit-flex: auto;
  -ms-flex: auto;
  flex: auto;
}

.c2 {
  position: relative;
}

.c0 {
  position: relative;
  background-color: black;
  width: 900px;
  max-width: 100%;
  padding: 16px;
}

.c0 .c1 {
  height: 100%;
}

.c0 .c4 {
  position: absolute;
  right: 0;
  bottom: 0;
  left: 0;
}

@media (min-width:120.0625em) {
  .c5 .c7 {
    font-size: 55px;
  }
}

@media (max-width:120em) {
  .c5 .c7 {
    font-size: 45px;
  }
}

@media (max-width:90em) {
  .c5 .c7 {
    font-size: 38px;
  }
}

@media (max-width:75.625em) {
  .c5 .c7 {
    font-size: 30px;
  }
}

@media (max-width:47.9375em) {
  .c5 .c7 {
    font-size: 20px;
  }
}

@media (max-width:75.625em) {
  .c0 {
    width: 75%;
  }
}

@media (max-width:47.9375em) {
  .c0 {
    width: 100%;
    padding: 10px;
  }
}

<div
  className="c0"
>
  <a
    className="test-class"
    href="#"
    otherFakeAttr="fake-value"
    title={undefined}
  >
    <div
      className="c1 c2"
    >
      <img
        className="c3"
        src={undefined}
      />
      <div
        className="c4 c5"
      >
        <img
          className="c6 "
          src="image/logo.png"
        />
        <div
          className="c7 c8"
        />
      </div>
    </div>
  </a>
</div>
`;
